src/main/java/com/github/manolo8/darkbot/core/manager/CrossApplicationEventManager.java

```diff
@@ -0,0 +1,169 @@
+package com.github.manolo8.darkbot.core.manager;
+
+import com.github.manolo8.darkbot.config.ConfigEntity;
+import eu.darkbot.api.PluginAPI;
+import eu.darkbot.api.extensions.Feature;
+import eu.darkbot.api.extensions.Installable;
+import eu.darkbot.api.extensions.Task;
+import eu.darkbot.api.managers.CrossApplicationAPI;
+import eu.darkbot.api.managers.EventBrokerAPI;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.SocketChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.UUID;
+
+
+@Feature(name = "Cross-application event", description = "Allow send events between applications")
+public class CrossApplicationEventManager implements Task, Installable, CrossApplicationAPI {
```

Comment:
This whole feature *really* feels like it's just a plugin feature that has been copy-pasted into the bot.

I believe the bot should offer a way to setup external communication that the plugins can use, but i don't think this is the right approach, because it takes the *whole* responsability of the system on the bot, which also makes it rather rigid. No option for 2 plugins to use this system at the same time for example.

Consider making this more of a "you can request a socket server be setup by the bot, and the bot takes care of it" but you're still the one who does everything else, meaning config for what addr to connect to, or the general layout of messaging. The bot simply provides the back-end for establishing the connection and handling multithreading if need be, but the messaging logic should be on the plugin side.

Reply:
Yes it was wrote as plugin and it can be run as plugin, but it required API PR https://github.com/darkbot-reloaded/DarkBotAPI/pull/92 for make possible general entry point from different places. 

It's only transport layer where exist mandatory type_id implement via UUID. It not have limitation for features that communicate over this channel so all plugin features can send messages in parallel and receive this messages known UUID it can be sure in structure before parsing it.

UUID is a marker show who send this message.

>This feature only send base64(UUID+message)+'\n' to router and generate OnMessage events when receive message in this format from another bot. It's all.

This idea is all that need to know developer of creating plugin code that work over this interface.

This code need in repo for has non-blocking connection that configured in single place.

Reply:
You make it sound like determining message format to be base64 only, and always prefixed by uuid, is not a limiting factor.

Again this is YOUR implementation that works for YOU, but the point of an API is to be able to be used by others too. If a plugin wants to establish a connection with an already-existing service, that communicates with a different scheme (eg: json) then what? This offers too little flexibility and it's just too-tailored to your specific use-case and your specific implementation decisions

Reply:
If plugin developer want to establish a connection with an already-existing service using low level messaging than he should know UUIDs of this service and listen responses from this service. This low level api need for establish a connection between feature instance in linked apps (not third-perty service it possible but it will required implementation a protocol of this service inside plugin). 
Message content not specified it can be json, xml, serialized java object - no matter. 
![image](https://github.com/darkbot-reloaded/DarkBot/assets/52008918/88d5c78b-df9b-43c1-9bc0-dbbcbd070161)

Now `this api + simple router` works like UDP broadcast. Where router address is a subnet for broadcast, UUID is a port, message is a datagram content. I not use bytes for message type because string easy parsing. because it easy parsing.
Over this api can be implemented RMI and RPC.
I don't agree that it has little flexibility.

Reply:
> If plugin developer want to establish a connection with an already-existing service using low level messaging than he should know UUIDs of this service

An already-existing service, will **not** have a uuid-prefixed message nor base64 encoding, nor be running in the same server as the other plugin.

Reply:
Why you focus on base64 format?, it no mater for plugin code side. it need only for debugging message sequence on router. And server is a router app in this model. Each bot is a client that ready to communicate. Using static UUID allow to find remote instances of feature like works ARP protocol in networking. There UUID is analog of MAC address. 

In this moment we not have any built-in services for communication and each developer can implement itself. it give freedom in building personal model of communication. Because I implemented only transport layer.

It not have limitation of communication between different features but it not prefer task. And it not have limitation for communicating between single app services but for this action need to create OnMessage event inside feature code.

Reply:
reverted from this PR, will implemented in another place